### ISSUE TITLE
Docs for testing package

### ISSUE THREAD

**Issue by sclasen** [sclasen/swfsm#26]
* how to use the test listener
* how to use the test interceptor
* how to use the stub FSM

**Comment by ryanwalls** [comment]
+1

**Comment by sclasen** [comment]
@ryanwalls lmk if the referenced PR is clear enough to get you started, or if there are particular things that need more filling in.

**Comment by ryanwalls** [comment]
@sclasen Thanks!  Yeah, that will get me started.  To this point I've been unit testing using the exposed `Tick` method and that has worked pretty well.  Though it is cumbersome to setup the last MarkerRecorded with StateData correctly.  But in general, I've been able to be TDD so far.   

Will try out this "integration" test approach soon and let you know if more examples would be helpful.

**Comment by sclasen** [comment]
Yeah, the TDD approach is where we started, but we got really sick of
manually creating all the right SWF response structs.  Tests with lots of
permutations took forever to get right and were hard to refactor.

To me doing it that way felt like the equivalent of mocking out a database
driver in a rails app, rather than just saying that 'this is a database
app'  and letting the database spit out all the right 'structs'

The one great thing about the TDD approach was the execution speed.  The
'integration' tests can get a bit long once you have a lot of them.
